File: 151003/Romanko/project/discussion/src/main/java/org/example/discussion/impl/comment/mapper/CommentMapper.java
package org.example.discussion.impl.comment.mapper;

import org.example.discussion.impl.comment.Comment;
import org.example.discussion.impl.comment.dto.CommentRequestTo;
import org.example.discussion.impl.comment.dto.CommentResponseTo;

import java.util.List;

public interface CommentMapper {
    CommentRequestTo commentToRequestTo(Comment comment);

    List<CommentRequestTo> commentToRequestTo(Iterable<Comment> comments);

    Comment dtoToEntity(CommentRequestTo commentRequestTo, String country);

    CommentResponseTo commentToResponseTo(Comment comment);

    List<CommentResponseTo> commentToResponseTo(Iterable<Comment> comments);
}

File: 151003/Barylko/Redis/Module.Discussion/Mappers/CommentMapper.cs
using Discussion.Dto.Request;
using Discussion.Dto.Response;
using Discussion.Models;
using Riok.Mapperly.Abstractions;
namespace Discussion.Mappers;

[Mapper]
public static partial class CommentMapper
{
    public static partial Comment ToEntity(this CommentRequestTo CommentRequestTo);
    public static partial CommentResponseTo ToResponse(this Comment Comment);
    public static partial IEnumerable<CommentResponseTo> ToResponse(this IEnumerable<Comment> Comments);

    public static Comment UpdateEntity(this Comment Comment, CommentRequestTo updateCommentRequestTo) => new()
    {
        Id = Comment.Id,
        IssueId = updateCommentRequestTo.IssueId,
        Content = updateCommentRequestTo.Content,
        Country = Comment.Country
    };
}
